test/types/string/methods/testDedent.chpl

```diff
@@ -0,0 +1,89 @@
+/* Temporary dummy method to make sure I can compile this test */
+proc string.dedent(columns=0, ignoreFirst=true) { return this; }
+
+var testStrings = [
+
+  // [0] Common use-case
+  """a
+     b
+     c""".dedent(),
+
+  // [1] Newlines before/after
+  """
+  a
+  b
+  c
+  """.dedent(),
+
+  // [2] Lines with only whitespace are normalized to a single newline.
+  // Note there is trailing whitespace in this example:
+  """  
+  a
+    
+  b
+  c
+  """.dedent(),
+
+  // [3] Remove 4 columns of indentation (there are 5 columns before b and c)
+  """a
+     b
+     c""".dedent(columns=4),
+
+  // [4] Removing (up to) 10 columns (ignoring first line though)
+  """ a
+        b
+         c
+          d
+           e""".dedent(columns=10),
```

Comment:
The default behavior for this example would also be interesting, as would the reverse of this example (diagonal top-right to bottom-left).  Maybe [7] is effectively the same thing, though?

Reply:
[7] is similar except the first line is empty. I can add one like this too.